File: frontend/src/types/notebooklm.ts
export interface NotebookSummary {
  id: string;
  title: string;
}

export interface ConnectNotebookRequest {
  notebook_id: string;
}

export interface ConnectNotebookResponse {
  project_id: string;
  notebook_id: string;
  notebook_title: string;
  connected: boolean;
}

export interface NotebookQueryRequest {
  query: string;
}

export interface NotebookQueryResponse {
  answer: string;
}

export interface ConnectedNotebook {
  notebook_id: string;
  notebook_title: string;
}

File: backend/app/schemas/notebooklm.py
from pydantic import BaseModel


class NotebookSummary(BaseModel):
    id: str
    title: str


class ConnectNotebookRequest(BaseModel):
    notebook_id: str


class ConnectNotebookResponse(BaseModel):
    project_id: str
    notebook_id: str
    notebook_title: str
    connected: bool


class NotebookQueryRequest(BaseModel):
    query: str


class NotebookQueryResponse(BaseModel):
    answer: str
